backend/infrahub/core/diff/enricher/labels.py

```diff
@@ -1,45 +1,198 @@
+            if property_diff.previous_value and not self._conflicts_only:
+                label = yield DisplayLabelRequest(
+                    node_id=property_diff.previous_value, branch_name=self.base_branch_name
+                )
+                if label:
+                    property_diff.previous_label = label
+            if property_diff.new_value and not self._conflicts_only:
+                label = yield DisplayLabelRequest(node_id=property_diff.new_value, branch_name=self.diff_branch_name)
+                if label:
+                    property_diff.new_label = label
+            if property_diff.conflict:
+                conflict_iterator = self._conflict_iterator(conflict_diff=property_diff.conflict)
+                label = None
+                try:
+                    while True:
+                        display_label_request = conflict_iterator.send(label)
+                        label = yield display_label_request
+                except StopIteration:
+                    ...
+
+    def _conflict_iterator(
+        self, conflict_diff: EnrichedDiffConflict
+    ) -> Generator[DisplayLabelRequest, str | None, None]:
+        if conflict_diff.base_branch_value:
+            label = yield DisplayLabelRequest(
+                node_id=conflict_diff.base_branch_value, branch_name=self.base_branch_name
+            )
+            if label:
+                conflict_diff.base_branch_label = label
+        if conflict_diff.diff_branch_value:
+            label = yield DisplayLabelRequest(
+                node_id=conflict_diff.diff_branch_value, branch_name=self.base_branch_name
+            )
+            if label:
+                conflict_diff.diff_branch_label = label
+
+    def _update_relationship_labels(self, enriched_diff: EnrichedDiffRoot) -> None:
+        for node in enriched_diff.nodes:
             if not node.relationships:
                 continue
-            node_schema = self.db.schema.get(name=node.kind, branch=diff_branch_name, duplicate=False)
+            node_schema = self.db.schema.get(name=node.kind, branch=self.diff_branch_name, duplicate=False)
             for relationship_diff in node.relationships:
                 relationship_schema = node_schema.get_relationship(name=relationship_diff.name)
                 relationship_diff.label = relationship_schema.label or ""
-                peer_kind = relationship_schema.peer
-                for element in relationship_diff.relationships:
-                    node_kind_map[peer_kind].append(element.peer_id)
 
-        display_label_map = await get_display_labels(db=self.db, nodes={diff_branch_name: node_kind_map})
-        for node in enriched_diff_root.nodes:
+    async def _get_display_label_map(
+        self, display_label_requests: set[DisplayLabelRequest]
+    ) -> dict[str, dict[str, str]]:
+        node_ids = [dlr.node_id for dlr in display_label_requests]
+        query = await NodeGetKindQuery.init(db=self.db, ids=node_ids)
```

Comment:
Not a blocker for this PR but it would be interesting to understand where these ids come from and see if we could collect the kind at the same time. In most cases, if we are querying the id from a query we should be able to get the kind at the same time.

Reply:
this can be addressed in IFC-584 (add peer kind to relationship element diffs) during the next round of diff updates